### ISSUE TITLE
Use the Reown dependency instead of WalletConnect

### ISSUE THREAD

**Issue by clauxx** [status-im/status-react#21336]
# Problem

Due to the WalletConnect rebranding to Reown, future updates will only be available under the new package names. The guide for upgrading is [here](https://docs.reown.com/walletkit/upgrade/from-web3wallet-react-native).

Make sure the older versions are supported, since we're stuck with the current one due to Android not working the up-to-date version.

## Acceptance Criteria

- [ ] Everything should work as it did before

**Comment by shivekkhurana** [comment]
We use the older version of WC sdk, because there was some issue with latest WC Sdk on Android. 

Check again if its fixed now.

**Comment by mohsen-ghafouri** [comment]
@shivekkhurana @clauxx could you please mention the exact error on Android? I upgraded it and all looks good just wanted to make sure I tested your scenario as well.

**Comment by clauxx** [comment]
> [@shivekkhurana](https://github.com/shivekkhurana) [@clauxx](https://github.com/clauxx) could you please mention the exact error on Android? I upgraded it and all looks good just wanted to make sure I tested your scenario as well.

@mohsen-ghafouri The scenario was that pairing wouldn't work on Android neither device (tried Android 14) nor emulator. Just nothing would happen when attempting to pair. I tried it now from develop with the new reown dependency and the issue with pairing persists for me. Did it work for you when you tested?

EDIT: seems like the issue is with the project id. I switched it with one I created and pairing works with it. Gotta figure why it doesn't work with the one we use.

EDIT2: fixed it by whitelisting the debug package name in the reown dashboard. All good 👍. They finally started giving out the actual error now `Fatal socket error: WebSocket connection closed abnormally with code: 3000 (Unauthorized: origin not allowed)`. Before it would just fail silently. (cc: @alwx finally this is fixed 🎆)
